src/backend/cpu/copy.cpp

```diff
@@ -63,6 +63,30 @@ namespace cpu
         return out;
     }
 
+    template<typename T>
+    void multiply_inplace(Array<T> &in, double val)
```

Comment:
tested it in standard c++ and it is working fine. Not sure if this will cause any issues in CUDA/OpenCL kernels.